presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -552,7 +553,8 @@ public TableStatisticsMetadata getStatisticsCollectionMetadata(ConnectorSession
         Set<ColumnStatisticMetadata> supportedStatistics = ImmutableSet.<ColumnStatisticMetadata>builder()
                 .addAll(hiveColumnStatistics)
                 // iceberg table-supported statistics
-                .addAll(super.getStatisticsCollectionMetadata(session, tableMetadata).getColumnStatistics())
+                .addAll(isCollectNativeSupportedStatistics(session) ?
```

Comment:
@nmahadevuni : ConnectorContext has boolean for native execution : context.getConnectorSystemConfig().isNativeExecution()

We could use this for disabling stats collection as native engine doesn't have support for it. There isn't a need to ask the user to flip this on or off I feel.